clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -837,13 +849,26 @@ invariant_checker_t::check_for_pc_discontinuity(
     std::string error_msg = "";
     bool have_cond_branch_target = false;
     addr_t cond_branch_target = 0;
-    const addr_t prev_instr_trace_pc = shard->prev_instr_.instr.addr;
-    if (prev_instr_trace_pc != 0 /*first*/ &&
+    memref_t prev_instr = shard->prev_instr_;
+    bool memref_is_kernel_event_marker = false;
+#ifdef UNIX
+    // Identify whether the current memref is a marker instead of an instruction. This
+    // will be handled differently.
+    if (memref.marker.type == TRACE_TYPE_MARKER &&
+        memref.marker.marker_type == TRACE_MARKER_TYPE_KERNEL_EVENT) {
+        memref_is_kernel_event_marker = true;
+        prev_instr = shard->last_instr_in_cur_context_;
+    }
+#endif
+    const addr_t current_memref_addr =
```

Comment:
(Optional, but recommended) Better to avoid this alias. It makes it a little harder to understand the conditions below. If we use either "memref.marker.marker_value" or "memref.instr.addr" directly, it's easily apparent what the condition expects the memref to be (a marker or an instr, respectively).